packages/lodestar-utils/src/json.ts

```diff
@@ -13,7 +16,8 @@ export function toJson(arg: unknown): Json {
       if (arg instanceof Uint8Array) return toHexString(arg);
       if (arg instanceof LodestarError) return toJson(arg.toObject());
       if (arg instanceof Error) return toJson(errorToObject(arg));
-      return arg as Json;
+      Object.entries(arg).forEach(([k, v]) => (copyArg[k] = toJson(v)));
+      return copyArg as Json;
```

Comment:
Warning! You are introducing recursion here, which has caused issues in the past. Object can have self-referencing properties which would create an infinite loop

Reply:
Unit tests, caught the recursion here https://github.com/ChainSafe/lodestar/pull/1755/checks?check_run_id=1393717117#step:8:149